### PR TITLE
test: add a size limit test

### PR DESCRIPTION
**Pull request by TheAlexLichter**
## Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Test coverage

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by aldarund** [comment]
I was thinking about such test, although wanted something that will show difference between new size vs old size :)

What the advantage loading page and getting js instead of getting sizes of *.js inside .nuxt/dist

**Comment by manniL** [comment]
@aldarund That'd be achievable with the current test as well, we'd only need to log the results. (Though they'd be hidden on CI anyway, so maybe an extra CI task would be good here).

Taking the `.js` files from the folder will result in roughly the same numbers (169.9 kB for the folder). But as they don't include the "initial page load"/the rendered HTML I decided to do it that way so we can catch potential size increases in base Nuxt components.

**Review by pi0 — APPROVED** [review]

File: test/unit/async-config.size-limit.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+describe('size-limit test', () => {
+  beforeAll(async () => {
+    const options = await loadFixture('async-config')
+    nuxt = new Nuxt(options)
+    port = await getPort()
+    await nuxt.server.listen(port, '0.0.0.0')
+
+    const { html } = await nuxt.server.renderRoute('/')
+    // Get all script URLs from the HTML
+    const $ = cheerio.load(html)
```
[the remark this answers is not in the document]
Comment (manniL) [reply test/unit/async-config.size-limit.test.js]: @pi0 Do you want me to rename the `const`? 🤔

**Comment by aldarund** [comment]
@manniL yes, if we have current size results we obv can compare them :) I saw it in react github 
https://github.com/facebook/react/pull/14119
But it dont work, their prev size always zero :)

**Comment by manniL** [comment]
@aldarund Ah, a bot! Nice idea
We could write our own for the scenario (which could be configurable/adaptable)  🤔

**Review by aldarund — APPROVED** [review]

**Pull request merged**